src/java/com/zimbra/ssdb/SSDBEphemeralStore.java

```diff
@@ -47,14 +47,28 @@
 public class SSDBEphemeralStore extends EphemeralStore {
     public static String SSDB_EPHEMERAL_STORE = "ssdb";
     private JedisPool pool;
-    public SSDBEphemeralStore(JedisPool pool) {
-        this.pool = pool;
+    private String url;
+    public SSDBEphemeralStore(String url) throws ServiceException {
+        this.url = url;
+        pool = getPool(url);
         setAttributeEncoder(new SSDBAttributeEncoder());
     }
+    private Jedis getResource() throws ServiceException {
+        Jedis jedis;
+        try{
+            jedis = pool.getResource();
+            jedis.ping();
+        } catch (JedisException e) {
+            pool.destroy();
+            pool = getPool(this.url);
+            jedis = pool.getResource();
+        }
+        return jedis;
+    }
     @Override
     public EphemeralResult get(EphemeralKey key, EphemeralLocation location) throws ServiceException {
         String encodedKey = encodeKey(key, location);
-        try (Jedis jedis = pool.getResource()) {
+        try (Jedis jedis = getResource()) {
             String encodedValue = jedis.get(encodedKey);
```

Comment:
Is there any scope for handling problems with the Jedis object here rather than in the new getResource() method?  Might limit extra network traffic to be only when there are problems.  Perhaps wrap the Jedis object in something that adds a layer of robustness instead of using it directly?  If things don't get better on the 2nd attempt, call wrapJedisException with the saved first exception.